server/src/main/java/org/apache/cassandra/sidecar/handlers/OpenApiHandler.java

```diff
@@ -0,0 +1,246 @@
+            
+            context.response()
+                   .putHeader("Content-Type", contentType)
+                   .end(openApiContent);
+        }
+        catch (Exception e)
+        {
+            logger.warn("Failed to load generated OpenAPI specification, falling back to basic config", e);
+            
+            // Fallback to basic OpenAPI config if generated file is not available
+            OpenAPI openAPI = createOpenApiFromConfig(openApiConfig);
+            String openApiJson = Json.pretty(openAPI);
+            
+            context.response()
+                   .putHeader("Content-Type", "application/json")
+                   .end(openApiJson);
+        }
+    }
+
+    /**
+     * Determines if the request is for YAML format based on path or Accept header
+     */
+    private boolean isYamlRequest(RoutingContext context)
+    {
+        String path = context.request().path();
+        if (path != null && path.endsWith(".yaml"))
+        {
+            return true;
+        }
+        
+        String acceptHeader = context.request().getHeader("Accept");
+        return acceptHeader != null && acceptHeader.contains("application/yaml");
+    }
+    
+    /**
+     * Loads the generated OpenAPI specification from resources or build output with caching
+     */
+    private String loadGeneratedOpenApiSpec(boolean isYaml) throws IOException
+    {
+        // First try to load from classpath resources (for packaged deployments)
+        String cached = isYaml ? cachedYamlSpec : cachedJsonSpec;
+        if (cached == null)
+        {
+            cached = loadFromClasspathResource(isYaml);
+            if (isYaml)
+            {
+                cachedYamlSpec = cached;
+            }
+            else
+            {
+                cachedJsonSpec = cached;
+            }
+        }
+        
+        if (cached != null)
+        {
+            return cached;
+        }
+
+        // Fallback to file system paths (for development)
+        return loadFromFileSystem(isYaml);
+    }
+
+    /**
+     * Attempts to load the OpenAPI spec from classpath resources
+     */
+    private String loadFromClasspathResource(boolean isYaml) throws IOException
+    {
+        String fileName = isYaml ? "openapi.yaml" : "openapi.json";
+        String resourcePath = "/openapi/" + fileName;
+        InputStream inputStream = getClass().getResourceAsStream(resourcePath);
+        if (inputStream == null)
+        {
+            return null;
+        }
+        
+        try
+        {
+            logger.debug("Loading OpenAPI specification from classpath: {}", resourcePath);
+            String content = new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+            logger.info("Loaded OpenAPI specification from classpath resource");
+            return content;
+        }
+        finally
+        {
+            inputStream.close();
+        }
```

Comment:
nit: use try-with-resources. 


```suggestion
        try (InputStream inputStream = getClass().getResourceAsStream(resourcePath))
        {
            if (inputStream == null)
            {
                return null;
            }

            logger.debug("Loading OpenAPI specification from classpath: {}", resourcePath);
            String content = new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
            logger.info("Loaded OpenAPI specification from classpath resource");
            return content;
        }
```